### PR TITLE
[SYCL] Perform SubgroupShuffle emulation for vectors of long long and half

### PR DESCRIPTION
**Pull request by maksimsab**

### CODE REVIEW

**Comment by maksimsab** [comment]
`Basic/memory-consumption.cpp` is flaky and has been removed here https://github.com/intel/llvm/pull/9063
Also it can be observed that `KernelAndProgram/kernel-bundle-get-kernel.cpp` and `Basic/kernel_info.cpp` are failing in other PRs as well.

Hi @bso-intel 
Could you please review this PR?

File: sycl/include/CL/sycl/version.hpp [hunk 1]
```diff
@@ -0,0 +1,12 @@
+//==------ version.hpp --- SYCL compiler version macro ---------*- C++ -*---==//
```
Comment (dm-vodopyanov) [inline sycl/include/CL/sycl/version.hpp:1]: This is auto-generated file which generates in `build` directory. It shouldn't be commited to the repo.

File: sycl/include/sycl/detail/spirv.hpp [hunk 2]
```diff
@@ -501,14 +501,20 @@ AtomicMax(multi_ptr<T, AddressSpace, IsDecorated> MPtr, memory_scope Scope,
...
+//   For double, long, long long, unsigned long, unsigned long long
+//   and half vectors we perform emulation with scalar version.
 // - The CUDA shfl intrinsics do not support vectors, and we use the _i32
 //   variants for all scalar types
 #ifndef __NVPTX__
 
 template <typename T>
 struct TypeIsProhibitedForShuffleEmulation
-    : bool_constant<std::is_same_v<vector_element_t<T>, double>> {};
+    : bool_constant<std::is_same_v<vector_element_t<T>, double> ||
```
Comment (dm-vodopyanov) [inline sycl/include/sycl/detail/spirv.hpp:512]: Instead you can create `type_list` of double, long, etc in generic_type_lists.cpp and use `is_contained` here. See examples here:

https://github.com/intel/llvm/blob/c1b2a1c7b07f869ade86a2446954f8ba636bc472/sycl/include/sycl/detail/generic_type_lists.hpp#L92
https://github.com/intel/llvm/blob/c1b2a1c7b07f869ade86a2446954f8ba636bc472/sycl/include/sycl/detail/generic_type_traits.hpp#L51

File: sycl/include/CL/sycl/version.hpp (hunk 1 above)
Comment (maksimsab) [reply sycl/include/CL/sycl/version.hpp]: oh, thanks. deleted.

**Review by bso-intel — APPROVED** [review]

**Review by dm-vodopyanov — CHANGES_REQUESTED** [review]
@maksimsab please investigate ABI test failure in pre-commit in check-sycl for Windows.

**Comment by maksimsab** [comment]
@dm-vodopyanov  I've found  that test in CUDA `Regression/same_unnamed_kernels.cpp` failed here as well: https://github.com/intel/llvm/actions/runs/4765868363/jobs/8472639817

So it isn't related to the change in this PR.

**Review by dm-vodopyanov — APPROVED** [review]
> @dm-vodopyanov  I've found  that test in CUDA `Regression/same_unnamed_kernels.cpp` failed here as well: https://github.com/intel/llvm/actions/runs/4765868363/jobs/8472639817
> 
> So it isn't related to the change in this PR.

@maksimsab please, create an issue in intel/llvm and add a tag "cuda"

**Comment by maksimsab** [comment]
@dm-vodopyanov Hi!
Previously mentioned regression has been fixed by somebody.
Can we merge this one?

**Pull request merged**